src/encoded/static/components/matrix_deeply_profiled_uniform_batch.js

```diff
@@ -0,0 +1,718 @@
+import PropTypes from 'prop-types';
+import QueryString from '../libs/query_string';
+import * as encoding from '../libs/query_encoding';
+import { Panel, PanelBody } from '../libs/ui/panel';
+import { svgIcon } from '../libs/svg-icons';
+import { tintColor, isLight } from './datacolors';
+import { DataTable } from './datatable';
+import * as globals from './globals';
+import { RowCategoryExpander, MATRIX_VISUALIZE_LIMIT } from './matrix';
+import { FacetList, SearchControls } from './search';
+import { CartAddAllElements } from './cart';
+
+
+/**
+ * Number of subcategory items to show when subcategory isn't expanded.
+ * @constant
+ */
+const SUB_CATEGORY_SHORT_SIZE = 0;
+
+/** Page name */
+const matrixName = 'deeply-profiled-uniform-batch-matrix';
+
+/**
+ * URL query string key for internal tag
+ */
+const internalTagKey = 'replicates.library.biosample.internal_tags';
+
+/**
+ * URL query string value for internal tag
+ */
+const internalTagValue = 'Deeply Profiled';
+
+/**
+ * Mapping of Assay type to assay titles
+ */
+const assayTypeToAssayTitles = [
+    { 'DNA binding': ['Histone ChIP-seq', 'TF ChIP-seq', 'Control ChIP-seq', 'Mint-ChIP-seq', 'Control Mint-ChIP-seq'] },
+    { 'RNA binding': ['RNA Bind-n-Seq', 'Control eCLIP', 'RIP-chip', 'eCLIP', 'iCLIP', 'RIP-seq', 'Switchgear'] },
+    { 'DNA accessibility': ['DNase-seq', 'FAIRE-seq', 'GM DNase-seq', 'ATAC-seq', 'MNase-seq'] },
+    { '3D chromatin structure': ['ChIA-PET', 'Hi-C', 'SPRITE', '5C'] },
+    { Transcription: ['polyA plus RNA-seq', 'microRNA counts', 'CRISPR RNA-seq', 'total RNA-seq', 'RAMPAGE', 'shRNA RNA-seq', 'small RNA-seq', 'siRNA RNA-seq', 'microRNA-seq', 'polyA minus RNA-seq', 'RNA microarray', 'BruUV-seq', 'CAGE', 'PAS-seq', 'RNA-PET', 'long read RNA-seq', 'BruChase-seq', 'Bru-seq', 'CRISPRi RNA-seq', 'PRO-seq', '5\' RLM RACE'] },
+    { 'DNA methylation': ['DNAme array', 'WGBS', 'RRBS', 'MeDIP-seq', 'MRE-seq', 'TAB-seq'] },
+    { 'Replication timing': ['Repli-chip', 'Repli-seq'] },
+    { Genotyping: ['genotyping array', 'DNA-PET'] },
+    { 'Single cell': ['scRNA-seq', 'snATAC-seq', 'long read scRNA-seq'] },
+    { 'DNA sequencing': ['WGS', 'Circulome-seq'] },
+    { 'RNA structure': ['icSHAPE', 'icLASER'] },
+    { Proteomics: ['MS-MS'] },
+];
+
+/**
+ * Sort a collection descendingly
+ * @param {object} a First item
+ * @param {Object} b Second item
+ *
+ * @returns Sorted collection
+ */
+const sortByDescending = (a, b) => {
+    if (a === b) {
+        return 0;
+    }
+    return a > b ? 1 : -1;
+};
+
+/** Page description */
+const matrixDescription = 'Cell line samples that were deeply profiled using a set of diverse biochemical approaches, listed with the corresponding batch identifiers.';
+
+/** Determines if user is on All-page or not  */
+const isAllDeeplyMatrix = (url) => url.includes('deeply-profiled-matrix');
+
+/** Switch page */
+const switchDeeplyProfilePageType = (isAll, pageUrl, navigate) => {
+    const query = new QueryString(pageUrl);
```

Comment:
The QueryString module is really designed to accept query strings only, not entire URIs. We usually use the `url` module to extract that. I’ve thought about modifying QueryString to accept either full URIs or query strings, but never felt it was difficult to extract the query string. But that avoids needing the call to `split()` below — just pass `query.format()` to that part of the string to `navigate`.